setzer/document/preview/preview_controller.py

```diff
@@ -122,6 +126,13 @@ def update_cursor(self, event):
                     link_target = _('Go to page ') + str(link[1].page_num)
                 break
 
+        if Gdk.BUTTON_MIDDLE in self.pressed_buttons:
+            cursor = self.cursor_move
+            adj = self.view.scrolled_window.get_hadjustment()
+            adj.set_value(adj.get_value() + self.panning_start_pos[0] - event.x)
+            adj = self.view.scrolled_window.get_vadjustment()
+            adj.set_value(adj.get_value() + self.panning_start_pos[1] - event.y)
```

Comment:
> These methods should't work with event objects at all.

In this case, should I create another method for this condition? Then I would create the class attribute `self.cursor_pos` and update it without making use of the event object and its attributes.